tools/ci_build/github/azure-pipelines/linux-cpu-minimal-build-ci-pipeline.yml

```diff
@@ -66,31 +78,59 @@ jobs:
               --minimal_build \
               --disable_exceptions
       workingDirectory: $(Build.SourcesDirectory)
+
   - task: CmdLine@2
-    displayName: Build minimal onnxruntime and run tests with exceptions enabled
+    displayName: Build minimal onnxruntime with exceptions enabled and type reduction disabled and run tests
     inputs:
       script: |
         docker run --rm \
           --volume $(Build.SourcesDirectory):/onnxruntime_src \
           --volume $(Build.BinariesDirectory):/build \
-          --volume $HOME/.test_data:/home/onnxruntimedev/.test_data \
+          --volume $(test_data_directory):/home/onnxruntimedev/.test_data \
           -e ALLOW_RELEASED_ONNX_OPSET_ONLY=1 \
           -e NIGHTLY_BUILD \
           -e BUILD_BUILDNUMBER \
           onnxruntimecentoscpubuild \
-            /bin/bash /onnxruntime_src/tools/ci_build/github/linux/ort_minimal/build_minimal_ort_and_run_tests.sh
+            /bin/bash /onnxruntime_src/tools/ci_build/github/linux/ort_minimal/build_minimal_ort_and_run_tests.sh \
+              --build-directory /build \
+              --reduced-ops-config /home/onnxruntimedev/.test_data/required_ops.ort_models.config
       workingDirectory: $(Build.SourcesDirectory)
+
+  - script: git checkout -- .
+    displayName: Discard local changes to Git repository files
+    workingDirectory: $(Build.SourcesDirectory)
+
+  - task: CmdLine@2
+    displayName: Build minimal onnxruntime with exceptions enabled and type reduction enabled and run tests
+    inputs:
+      script: |
+        docker run --rm \
+          --volume $(Build.SourcesDirectory):/onnxruntime_src \
+          --volume $(Build.BinariesDirectory):/build \
+          --volume $(test_data_directory):/home/onnxruntimedev/.test_data \
+          -e ALLOW_RELEASED_ONNX_OPSET_ONLY=1 \
+          -e NIGHTLY_BUILD \
+          -e BUILD_BUILDNUMBER \
+          onnxruntimecentoscpubuild \
+            /bin/bash /onnxruntime_src/tools/ci_build/github/linux/ort_minimal/build_minimal_ort_and_run_tests.sh \
+              --build-directory /build/with_type_reduction \
+              --reduced-ops-config /home/onnxruntimedev/.test_data/required_ops_and_types.ort_models.config \
```

Comment:
Do we need to throw away the build between having type reduction disabled and enabled, or can we just do an incremental build to save time? Not sure if running via docker supports that though.

Reply:
it should be possible to do an incremental build. however, the test report files will have the same name, so need to upload them twice or somehow not overwrite them. one of these reduced minimal build steps takes about 4 minutes, so the time savings would not be huge.
IMO, i'd prefer doing a clean build for simplicity. unless we specifically want to test switching the type reduction on with an existing build directory/repo.

---
In reply to: [573278739](https://github.com/microsoft/onnxruntime/pull/6622#discussion_r573278739) [](ancestors = 573278739)